File: js/src/Formation/FormationPointModel.js
/**
 * @class FormationPointModel
 */
export default class FormationPointModel {
    constructor(level = -1, row = -1, column = -1) {
        if (level === -1 || column === -1 || row === -1) {
            throw new TypeError('Invalid parameters. Expected a positive integer');
        }

        this.level = level;
        this.row = row;
        this.column = column;
    }

    /**
     * @for FormationPointModel
     * @param comparePoint {FormationPointModel|array}
     * @return {boolean}
     */
    isEqualToComparePoint(comparePoint) {
        return comparePoint.level === this.level &&
            comparePoint.row === this.row &&
            comparePoint.column === this.column;
    }

    /**
     * @for FormationPointModel
     * @method toArray
     * @return {array}
     */
    toArray() {
        return [this.level, this.row, this.column];
    }
}

File: cs/FormationPointModel.cs
using System;
using Sytem.Collections;

public class FormationPointModel
{
    public int level;
    public int row;
    public int column;

    public FormationPointModel(int level, int row, int column)
    {
        level = level;
        row = row;
        column = column;
    }

    public boolean isEqualToComparePoint(FormationpointModel comparePoint)
    {
        return comparePoint.level === level &&
            comparePoint.row === row &&
            comparePoint.column === column;
    }

    public int[] toArray()
    {
        const arr = new int[] {level, row, column};

        return arr;
    }
}
